src/Dotenv.cs

```diff
@@ -55,7 +62,18 @@ public static Dotenv Load(Stream input)
         /// <returns>Dictionary of environment variables.</returns>
         public Dictionary<string, string> GetVariables()
         {
-            return variables;
+            // Return a copy so caller cannot modify internal state.
+            return CloneVariables( );
+        }
+
+        private Dictionary<string, string> CloneVariables()
```

Comment:
Just to check, do you mean for CloneVariables() ?  I had a comment where it was called but I can add another at the function def if you believe it will help.